homeassistant/components/radarr/sensor.py

```diff
@@ -95,154 +76,137 @@
-                attributes[data.path] = "{:.2f}/{:.2f}{} ({:.2f}%)".format(
-                    free_space,
-                    total_space,
-                    self.native_unit_of_measurement,
-                    0 if total_space == 0 else free_space / total_space * 100,
-                )
-        elif self.entity_description.key == "movies":
-            return {to_key(movie): movie.hasFile for movie in self.data}
+            return self.get_strings(cast(str, self.native_unit_of_measurement))
+        if self.entity_description.key == "movies":
+            return {to_key(movie): movie.hasFile for movie in self.coordinator.movies}
         if self.entity_description.key == "status":
-            return self.data.attributes
-        return attributes
-
-    async def async_update(self) -> None:
-        """Update the data for the sensor."""
-        try:
-            if self.entity_description.key == "diskspace":
-                data = await self.radarr.async_get_diskspace()
-                if len(self.included) > 0:
-                    self.data = [datum for datum in data if datum.path in self.included]
-                else:
-                    self.data = data
-                space = sum(mount.freeSpace for mount in self.data)
-                self._attr_native_value = (
-                    f"{to_unit(space, self.native_unit_of_measurement):.2f}"
-                )
-            if self.entity_description.key == "upcoming":
-                local = dt_util.start_of_local_day().replace(microsecond=0)
-                start = dt_util.as_utc(local)
-                end = start + timedelta(days=self.days)
-                self.data = await self.radarr.async_get_calendar(
-                    start_date=start, end_date=end
-                )
-                self._attr_native_value = len(self.data)
-            if self.entity_description.key == "commands":
-                self.data = await self.radarr.async_get_commands()
-                self._attr_native_value = len(cast(list, self.data))
-            if self.entity_description.key == "status":
-                self.data = await self.radarr.async_get_system_status()
-                self._attr_native_value = self.data.version
-            if self.entity_description.key == "movies":
-                self.data = await self.radarr.async_get_movies()
-                self._attr_native_value = len(cast(list, self.data))
-        except ArrException as ex:
-            _LOGGER.warning(ex)
-        self._attr_available = self.native_value is not None
-
-
-def release_date(movie: RadarrMovie) -> str:
-    """Get release date."""
-    date = movie.physicalRelease or movie.digitalRelease or movie.inCinemas
-    return date.isoformat()
-
-
-def to_key(movie: RadarrMovie):
+            return self.coordinator.system_status.attributes
+        return {to_key(movie): movie.releaseDate for movie in self.coordinator.calendar}
```

Comment:
This is the default for a lot of sensors and it should be the attributes for none. This should be moved to to be calendar platform.

Reply:
This reflects the old behavior. Since we are not merging into dev directly, I can remove this for now and have a new PR just for the calendar.